tests/tensorflow/test_tensorflow2_autolog.py

```diff
@@ -41,8 +41,12 @@
 
 @pytest.fixture(autouse=True)
 def clear_session():
-    yield
-    tf.keras.backend.clear_session()
+    try:
+        yield
+    finally:
+        # Shut down async logging to ensure we terminate thread pools.
+        mlflow.shut_down_async_logging()
+        tf.keras.backend.clear_session()
```

Comment:
We normally don't need try-finally in a pytest fixture. pytest handles errors.

Reply:
Ah ok!